src/main/kotlin/com/cognifide/gradle/aem/common/instance/LocalInstance.kt

```diff
@@ -19,109 +17,88 @@ import java.io.FileFilter
 
 @Suppress("TooManyFunctions")
 class LocalInstance(aem: AemExtension) : Instance(aem) {
+    init {
+        user.set(USER)
+    }
 
-    override var user: String = USER
-
-    var debugPort: Int = 5005
+    val debugPort = common.obj.int { convention(5005) }
 
-    var debugAddress: String = ""
+    val debugAddress = common.obj.string {}
 
-    var openPath: String = "/"
+    val openPath = common.obj.string { convention("/") }
 
-    val httpOpenUrl get() = when (openPath) {
-        "/" -> httpUrl
-        else -> "${httpUrl}$openPath"
+    val httpOpenUrl get() = when (openPath.get()) {
+        "/" -> httpUrl.get()
+        else -> "${httpUrl.get()}${openPath.get()}"
     }
 
-    private val debugSocketAddress: String get() = when (debugAddress) {
-        "*" -> "0.0.0.0:$debugPort"
-        "" -> "$debugPort"
-        else -> "$debugAddress:$debugPort"
-    }
+    val jvmOpts = common.obj.strings { convention(listOf("-server", "-Xmx2048m", "-XX:MaxPermSize=512M", "-Djava.awt.headless=true")) }
 
-    @get:JsonIgnore
-    val jvmOptsDefaults: List<String> get() = mutableListOf<String>().apply {
-        if (debugPort in 1..65535) {
-            add(jvmDebugOpt)
-        }
-    }
+    val jvmAgentOpt: String? get() = jvmAgents.get().joinToString(" ") { "-javaagent:$it" }.ifBlank { null }
 
-    @get:JsonIgnore
-    private val jvmDebugOpt: String get() = when {
-        localManager.javaLauncher.get().metadata.languageVersion >= JavaLanguageVersion.of(9) ->
-            "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$debugSocketAddress"
-        else ->
-            "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$debugPort"
+    @Suppress("MagicNumber")
+    val jvmDebugOpt: String? get() = when (debugPort.orNull) {
+        in 1..65535 -> {
+            val address = when {
+                localManager.javaLauncher.get().metadata.languageVersion >= JavaLanguageVersion.of(9) -> {
+                    when {
+                        debugAddress.orNull == "*" -> "0.0.0.0:${debugPort.get()}"
+                        debugAddress.orNull.isNullOrBlank() -> "${debugPort.get()}"
+                        else -> "${debugAddress.orNull}:${debugPort.get()}"
+                    }
+                }
+                else -> {
+                    "${debugPort.get()}"
+                }
+            }
+            "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=$address"
+        }
+        else -> null
     }
 
-    @get:JsonIgnore
-    var jvmOpts: List<String> = listOf(
-        "-server", "-Xmx2048m", "-XX:MaxPermSize=512M", "-Djava.awt.headless=true"
-    )
+    val jvmOptsString: String get() = (jvmOpts.get() + jvmDebugOpt + jvmAgentOpt).filterNot { it.isNullOrBlank() }.joinToString(" ")
```

Comment:
clue of change is to have it also `jvmAgentOpt` which will auto resolve files just before creating AEM instances